src/main/scala/org/broadinstitute/dsde/vault/model/Generic.scala

```diff
@@ -0,0 +1,112 @@
+package org.broadinstitute.dsde.vault.model
+
+import com.wordnik.swagger.annotations.{ApiModel, ApiModelProperty}
+import spray.json.DefaultJsonProtocol
+import scala.annotation.meta.field
+
+object GenericJsonProtocol extends DefaultJsonProtocol {
+  implicit val impGenericSysAttrs = jsonFormat5(GenericSysAttrs)
+  implicit val impGenericEntity = jsonFormat6(GenericEntity)
+  implicit val impGenericRelationship = jsonFormat2(GenericRelationship)
+  implicit val impGenericRelEnt = jsonFormat2(GenericRelEnt)
+  implicit val impBlobSpecification = jsonFormat2(BlobSpecification)
+  implicit val impGenericEntityIngest = jsonFormat4(GenericEntityIngest)
+  implicit val impGenericRelationshipIngest = jsonFormat4(GenericRelationshipIngest)
+  implicit val impGenericIngest = jsonFormat2(GenericIngest)
+  implicit val impGenericAttributeSpec = jsonFormat2(GenericAttributeSpec)
+  implicit val impGenericQuery = jsonFormat3(GenericEntityQuery)
+}
+
+@ApiModel("system-generated entity attributes")
```

Comment:
Would it be possible to move all/most of the model classes into vault-common, if they are shared between DM and API? I realize API has models that DM doesn't, such as BlobSpecification.